File: game_dice.py
# game_dice.py
import random
import asyncio
from db_manager import get_inventory, update_user_points

COMMAND = "نرد"

async def handle(bot, message):
    if message.text != COMMAND:
        return

    uid = message.from_user.id
    inventory = await get_inventory(uid)

    # رسالة ترحيب مؤقتة
    await bot.send_message(message.chat.id, "🎲 جاري رمي نرد الإمبراطورية...\n⚔️ ركّز، فالحظ لا يبتسم مرتين.")

    # "رمي النرد"
    value = random.randint(1, 6)

    # تأثير درامي
    await asyncio.sleep(3.5)

    result_text = ""
    points_change = 0

    if value >= 5:
        # فوز إمبراطوري
        points_change = 20
        if "سيف الإمبراطور 🔱" in inventory:
            points_change = int(points_change * 1.2)
        if "قفاز القوة" in inventory:
            points_change = int(points_change * 1.25)
        if "خاتم الحظ 💍" in inventory and random.random() < 0.1:
            points_change += 5

        new_points = await update_user_points(uid, points_change)
        result_text = f"""
┏━━━━━━━ ● ━━━━━━━┓
         ⌯ فـوز إمـبـراطـوري ⌯
┗━━━━━━━ ● ━━━━━━━┛

🔥 الحظ يبتسم لك : [ {value} ]
🏆 النقاط المكتسبة : +{points_change}
✨ رصيدك الحالي : {new_points} نقاط
        """

    elif value >= 3:
        # حظ متوسط
        points_change = 10
        if "سيف الإمبراطور 🔱" in inventory:
            points_change = int(points_change * 1.2)
        if "قفاز القوة" in inventory:
            points_change = int(points_change * 1.25)

        new_points = await update_user_points(uid, points_change)
        result_text = f"""
┏━━━━━━━ ● ━━━━━━━┓
         ⌯ حظ متوسط ⌯
┗━━━━━━━ ● ━━━━━━━┛

🔥 الحظ يبتسم لك : [ {value} ]
🏆 النقاط المكتسبة : +{points_change}
✨ رصيدك الحالي : {new_points} نقاط
        """

    else:
        # خسارة
        points_change = -5
        if "درع الحصن 🛡️" in inventory:
            points_change = int(points_change * 0.5)
        if "عباءة الظلال 🧥" in inventory:
            points_change = int(points_change * 0.8)

        new_points = await update_user_points(uid, points_change)
        result_text = f"""
┏━━━━━━━ ● ━━━━━━━┓
         ⌯ خسارة ساحقة ⌯
┗━━━━━━━ ● ━━━━━━━┛

💀 الحظ : [ {value} ]
💸 خسارتك : {abs(points_change)} نقاط
✨ رصيدك الحالي : {new_points} نقاط
        """

    # إرسال النتيجة النهائية
    await bot.send_message(message.chat.id, result_text)

File: game_dice.js
// game_dice.js
const db = require('./db_manager');

const COMMAND = "نرد";
const MISSION_TYPE = "dice";       // نوع المهمة في db_manager
const REWARD_ITEM = "🎁 صندوق الحظ النادر";

async function handle(bot, ctx) {
    if (ctx.message.text !== COMMAND) return;

    const uid = ctx.from.id;
    const inventory = await db.get_inventory(uid);

    // رسالة ترحيب مؤقتة
    await ctx.reply("🎲 جاري رمي نرد الإمبراطورية...\n⚔️ ركّز، فالحظ لا يبتسم مرتين.");

    try {
        // إرسال نرد Telegram
        const diceMsg = await ctx.sendDice();
        const value = diceMsg.dice.value;

        // تأثير درامي
        await new Promise(r => setTimeout(r, 3500));

        let resultText = "";
        let prizeOrPenalty = 0;

        if (value >= 5) {
            // فوز إمبراطوري
            prizeOrPenalty = 200;
            if (inventory.includes("سيف الإمبراطور 🔱")) prizeOrPenalty = Math.floor(prizeOrPenalty * 1.2);
            if (inventory.includes("قفاز القوة")) prizeOrPenalty = Math.floor(prizeOrPenalty * 1.25);
            if (inventory.includes("خاتم الحظ 💍") && Math.random() < 0.1) prizeOrPenalty += 50;

            const newGold = await db.update_user_gold(uid, prizeOrPenalty);
            resultText = `
┏━━━━━━━ ● ━━━━━━━┓
         ⌯ فـوز إمـبـراطـوري ⌯
┗━━━━━━━ ● ━━━━━━━┛

🔥 الحظ يبتسم لك : [ ${value} ]
💰 الجائزة : +${prizeOrPenalty} ذهب
✨ رصيدك الحالي : ${newGold} ذهب
            `;
        } else if (value >= 3) {
            // حظ متوسط
            prizeOrPenalty = 50;
            if (inventory.includes("سيف الإمبراطور 🔱")) prizeOrPenalty = Math.floor(prizeOrPenalty * 1.2);
            if (inventory.includes("قفاز القوة")) prizeOrPenalty = Math.floor(prizeOrPenalty * 1.25);

            const newGold = await db.update_user_gold(uid, prizeOrPenalty);
            resultText = `
┏━━━━━━━ ● ━━━━━━━┓
         ⌯ حظ متوسط ⌯
┗━━━━━━━ ● ━━━━━━━┛

🔥 الحظ يبتسم لك : [ ${value} ]
💰 الجائزة : +${prizeOrPenalty} ذهب
✨ رصيدك الحالي : ${newGold} ذهب
            `;
        } else {
            // خسارة
            prizeOrPenalty = -30;
            if (inventory.includes("درع الحصن 🛡️")) prizeOrPenalty = Math.floor(prizeOrPenalty * 0.5);
            if (inventory.includes("عباءة الظلال 🧥")) prizeOrPenalty = Math.floor(prizeOrPenalty * 0.8);

            const newGold = await db.update_user_gold(uid, prizeOrPenalty);
            resultText = `
┏━━━━━━━ ● ━━━━━━━┓
         ⌯ خسارة ساحقة ⌯
┗━━━━━━━ ● ━━━━━━━┛

💀 الحظ : [ ${value} ]
💸 خسارتك : ${Math.abs(prizeOrPenalty)} ذهب
✨ رصيدك الحالي : ${newGold} ذهب
            `;
        }

        // إرسال النتيجة النهائية
        await ctx.reply(resultText);

        // تحقق من المهمة اليومية
        const missionCompleted = await db.complete_mission(uid, MISSION_TYPE);
        if (missionCompleted) {
            await db.add_to_inventory(uid, REWARD_ITEM);
            await ctx.reply(`✅ تم إكمال المهمة اليومية!\n🎁 تم إضافة ${REWARD_ITEM} إلى مخزونك`);
        }

    } catch (err) {
        console.error(err);
        await ctx.reply("❌ فشل في رمي النرد، حاول مرة أخرى");
    }
}

module.exports = { handle };
